packages/ts/generator-typescript-plugin-model/src/ModelSchemaProcessor.ts

```diff
@@ -0,0 +1,200 @@
+    const schemaPath = convertReferenceSchemaToPath(schema);
+    const typeName = convertReferenceSchemaToSpecifier(schema);
+    const typePath = this.#dependencies.paths.createRelativePath(schemaPath, this.#cwd);
+    const modelPath = this.#dependencies.paths.createRelativePath(`${schemaPath}Model`, this.#cwd);
+    const modelName = `${typeName}Model`;
+    const refType =
+      this.#dependencies.imports.default.getIdentifier(typePath) ??
+      this.#dependencies.imports.default.add(typePath, typeName, true);
+    const type = ts.factory.createTypeReferenceNode(refType);
+    const model =
+      this.#dependencies.imports.default.getIdentifier(modelPath) ??
+      this.#dependencies.imports.default.add(modelPath, modelName);
+    const modelType = ts.factory.createTypeReferenceNode(model);
+    return [type, modelType, model, []];
+  }
+
+  #processArray(schema: ArraySchema): [TypeNode, TypeNode, Identifier, Expression[]] {
+    const model = this.#getBuiltinFormExport('ArrayModel');
+    const [itemType, itemModelType, itemModel, itemArgs] = new ModelSchemaProcessor(
+      schema.items,
+      this.#dependencies,
+      this.#cwd,
+    ).process();
+    const type = ts.factory.createTypeReferenceNode(ts.factory.createIdentifier('ReadonlyArray'), [itemType]);
+    const modelType = ts.factory.createTypeReferenceNode(model, [itemType, itemModelType]);
+    return [type, modelType, model, [itemModel, itemArgs]];
+  }
+
+  #processRecord(schema: MapSchema): [TypeNode, TypeNode, Identifier, Expression[]] {
+    const model = this.#getBuiltinFormExport('ObjectModel');
+
+    let valueType: TypeNode;
+    if (typeof schema.additionalProperties === 'boolean') {
+      valueType = ts.factory.createKeywordTypeNode(ts.SyntaxKind.AnyKeyword);
+    } else {
+      const valueSchema: Schema = schema.additionalProperties;
+      [valueType] = new ModelSchemaProcessor(valueSchema, this.#dependencies, this.#cwd).process();
+    }
+
+    const type = ts.factory.createTypeReferenceNode(ts.factory.createIdentifier('Record'), [
+      ts.factory.createKeywordTypeNode(ts.SyntaxKind.StringKeyword),
+      valueType,
+    ]);
+    const modelType = ts.factory.createTypeReferenceNode(model, [type]);
+    return [type, modelType, model, []];
+  }
+
+  #getValidators(schema: AnnotatedSchema): Expression[] {
+    const validators: Expression[] = [];
+    schema['x-annotations'].forEach((annotationText: string) => {
+      const annotation = parseAnnotation(annotationText);
+      const validator = this.#getBuiltinFormExport(annotation.simpleName);
+      const newValidator = ts.factory.createNewExpression(
+        validator,
+        undefined,
+        annotation.arguments.map(convertAnnotationArgumentToExpression),
+      );
+      validators.push(newValidator);
+    });
+    return validators;
+  }
+
+  #getBuiltinFormExport(specifier: string): Identifier {
+    const modelPath = this.#dependencies.paths.createBareModulePath('@vaadin/form', false);
```

Comment:
It would probably be more readable if you add dependencies decomposition at start:
```ts
const { imports, paths } = this.#dependencies
```